### DISCUSSION TITLE
Pre-fill and parallel multi-segment mode are finally here

### DISCUSSION THREAD

**Discussion by awawa-dev** [awawa-dev/HyperHDR#497] [Show and tell]
The latest HyperSerialESP32 and HyperSPI development versions introduce new features, offering a pre-fill mode to further improve sequential rendering of LEDs, and something previously unattainable for sk6812 RGBW LEDs: parallel rendering of LED segments, ensuring perfect synchronization between them. Thanks to this, you can easily build an LED system, e.g. for television, doubling the refresh rate without losing quality and without delays between segments. In this way, we can bypass the rather slow Neopixel data bus and practically forget about SPI RGB LED strips (APA102 clones, sk9822 etc.) with poor colors, whose only advantage is fast data transmission. You can read more here: https://github.com/awawa-dev/HyperSerialESP32/pull/17

You can compile them online (you can find manual in the projects' wiki) or download them from the recent Github Action.
Release:
https://github.com/awawa-dev/HyperSerialESP32/releases/tag/v9.0.0.0
https://github.com/awawa-dev/HyperSPI/releases/tag/v9.0.0.0

**Comment by Cptn-Reynolds** [comment]
Does parallel multi-segment mode already make sense for a 4m strip on a 55" ambilight project or are we talking much longer LED strips here? :)

**Reply by awawa-dev** [reply]
Multisegment is always faster, the longer the LED strip, the more noticeable it is. And the considerations when it is worth/need to use it and the explanation of it you can find here: https://www.hyperhdr.eu/2023/02/ultimate-guide-on-how-to-build-led.html#chapter5

**Reply by xSterney** [reply]
> Does parallel multi-segment mode already make sense for a 4m strip on a 55" ambilight project or are we talking much longer LED strips here? :)

I have a 55" TV too and for me the normal wiring is already more than fast enough using hyperspi. 
I wasn´t happy connecting the rpi to the esp via wled (wifi) because it wasnt fast enough, but with hyperspi im very happy

**Reply by Cptn-Reynolds** [reply]
well, there is always faster, isn't there =D but yeah, in my setup this would mean 9ms instead of 4,5ms and I really don't think this would be noticable to me. Still good to know, that there are improvement options for those who need them

 @xSterney so whats your current setup? ESP32 via WiFi?

**Reply by xSterney** [reply]
> well, there is always faster, isn't there =D but yeah, in my setup this would mean 9ms instead of 4,5ms and I really don't think this would be noticable to me. Still good to know, that there are improvement options for those who need them
> 
> @xSterney so whats your current setup? ESP32 via WiFi?

no as i said i run it with hyperspi now :D
I have the ezcoo matrix simple connected to my ms2130.
And the rpi 3b+ is connected via hyperspi with the esp32 which drives the sk6812nw.
Very happy with the setup and everything works fine

**Reply by Cptn-Reynolds** [reply]
Yeah but how does the rpi communicate to esp32? Are those directly connected via GPIO ports or whats the physical (or wireless) connection?

**Reply by xSterney** [reply]
ah now i know what you mean.
HyperSPI only runs via gpio ports.
